addons/crm/static/src/js/crm_kanban.js

```diff
@@ -4,12 +4,156 @@
      * This Kanban Model make sure we display a rainbowman
      * message when a lead is won after we moved it in the
      * correct column and when it's grouped by stage_id (default).
+     * Apart from that, in the KanbanColumnProgressBar, we add
+     * support for showing sum of the expected MRR (after the regular
+     * sum field) if the logged user has the enough rights.
      */
 
+    import KanbanColumn from 'web.KanbanColumn';
+    import KanbanColumnProgressBar from 'web.KanbanColumnProgressBar';
     import KanbanModel from 'web.KanbanModel';
+    import KanbanRenderer from 'web.KanbanRenderer';
     import KanbanView from 'web.KanbanView';
+    import utils from 'web.utils';
     import viewRegistry from 'web.view_registry';
 
+    const CrmKanbanColumnProgressBar = KanbanColumnProgressBar.extend({
+
+        /**
+         * @constructor
+         * @override
+         */
+        init(parent, options, columnState) {
+            this._super.apply(this, arguments);
+            this.recurringRevenueSumField = columnState.progressBarValues.recurring_revenue_sum_field;
+            // Check whether the given MRR sum field is valid or not
+            this.isRecurringRevenueSumFieldValid = Object.keys(columnState.fields).includes(columnState.progressBarValues.recurring_revenue_sum_field);
+            this.recurringRevenueSumFieldLabel = this.isRecurringRevenueSumFieldValid ? columnState.fields[this.recurringRevenueSumField].string : false;
+            // Previous progressBar state
+            const state = options.progressBarStates[this.columnID];
+            if (state) {
+                this.totalRecurringRevenue = state.totalRecurringRevenue;
+            }
+        },
+
+        /**
+         * @override
+         */
+        willStart() {
+            const userHasGroup = this.getSession().user_has_group('crm.group_use_recurring_revenues')
+                .then((useRecurringRevenues) => {
+                    // only show MRR related info if 'Recurring Revenues' is enabled, and provided MRR sum field is valid
+                    this.showRecurringRevenue = useRecurringRevenues && this.isRecurringRevenueSumFieldValid;
+                });
+            return Promise.all([this._super(...arguments), userHasGroup]);
+        },
+
+        /**
+         * @override
+         */
+        start() {
+            this._super.apply(this, arguments);
```

Comment:
Should take care of super as start is an important part of the widget life cycle